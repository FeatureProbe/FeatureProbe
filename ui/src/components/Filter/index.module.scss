.popup {
  padding: 8px 0 !important;
  margin-top: 4px !important;
  min-width: 180px !important;
}

.popup-footer {
  border-top: 1px solid #e2e5e9;
  text-align: right;
  padding-top: 7px;
  padding-right: 12px;
}

.popup-footer-clear {
  font-size: 13px;
  color: #353a40;
  cursor: pointer;
}

.popup-footer-confirm {
  font-size: 13px;
  color: #556ee6;
  cursor: pointer;
  margin-left: 12px;
}

.icon-filter {
  color: #919aac;
  margin-left: 8px;
  position: relative;
  top: 1px;
  cursor: pointer;
}

.icon-filter-selected {
  color: #556ee6;
}