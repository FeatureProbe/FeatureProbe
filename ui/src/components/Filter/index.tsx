import { ReactElement, SyntheticEvent, useState } from 'react';
import { Popup } from 'semantic-ui-react';
import { FormattedMessage } from 'react-intl';
import Icon from 'components/Icon';

import styles from './index.module.scss';

interface IProps {
  handleConfirm(): void
  handleClear(): void
  selected?: boolean;
  children: ReactElement
}

const Filter = (props: IProps) => {
  const { selected, children, handleConfirm, handleClear } = props;
  const [ popupOpen, savePopupOpen ] = useState<boolean>(false);

  return (
    <Popup
      basic
      open={popupOpen}
      on='click'
      position='bottom right'
      className={styles.popup}
      trigger={
        <Icon 
          type='filter' 
          customclass={`${styles['icon-filter']} ${selected && styles['icon-filter-selected']}`} 
          onClick={(e: SyntheticEvent) => {
            document.body.click();
            e.stopPropagation();
            savePopupOpen(true);
          }} 
        />
      }
    >
      <div onClick={(e) => e.stopPropagation()}>
        {children}
        <div className={styles['popup-footer']}>
          <span className={styles['popup-footer-clear']} 
            onClick={() => { 
              handleClear();
            }}
          >
            <FormattedMessage id='common.clear.text' />
          </span>
          <span 
            className={styles['popup-footer-confirm']} 
            onClick={() => {
              handleConfirm();
              savePopupOpen(false);
            }}
          >
            <FormattedMessage id='common.confirm.text' />
          </span>
        </div>
      </div>
    </Popup>
  );
};

export default Filter;



