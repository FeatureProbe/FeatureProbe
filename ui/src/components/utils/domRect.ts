export class DOMRect {
    x: number;
    width: number;
    y: number;
    height: number;
    constructor(x: number, y: number, width: number, height: number) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
    }
}