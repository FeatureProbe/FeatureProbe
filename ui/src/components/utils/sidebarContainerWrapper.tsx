import { SidebarContainer } from 'layout/hooks';

export default SidebarContainer.Provider;
