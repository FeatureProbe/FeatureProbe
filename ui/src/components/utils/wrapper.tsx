import en_US from '../../locales/en-US.json';
import { IntlProvider } from 'react-intl';

export const IntlWrapper: React.FC = ({ children }) => {
  return (
    <IntlProvider locale="en" messages={en_US}>{children}</IntlProvider>
  );
};
