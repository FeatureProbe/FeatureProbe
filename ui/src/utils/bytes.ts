export function getBytes(str: string) {
  let bytes = str.length,
    i = 0;
  for (; i < bytes; i++) if (str.charCodeAt(i) > 255) bytes++;

  return bytes;
}

const MB = 1048576;
export const MAX_SIZE = MB;