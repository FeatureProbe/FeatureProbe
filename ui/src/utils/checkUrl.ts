function escapeStringRegexp(string: string): string {
  if (typeof string !== 'string') {
    throw new TypeError('Expected a string');
  }

  return string
    .replace(/[|\\{}()[\]^$+*?.]/g, '\\$&')
    .replace(/-/g, '\\x2d');
}

export function matchUrl(matcher: string, originUrl: string, targetUrl: string): boolean {
  let href = '';
  let hash = '';
  let search = '';

  try {
    const fullUrl = new URL(targetUrl, '');
    href = fullUrl.href;
    hash = fullUrl.hash;
    search = fullUrl.search;
  } catch (error) {
    href = targetUrl;
  }

  let regex;
  let testUrl;

  switch (matcher) {
    case 'EXACT':
      testUrl = href;
      regex = new RegExp('^' + escapeStringRegexp(originUrl) + '/?$');
      break;
    case 'SIMPLE':
      testUrl = href.replace(hash, '').replace(search, '');
      regex = new RegExp('^' + escapeStringRegexp(originUrl) + '/?$');
      break;
    case 'SUBSTRING':
      testUrl = href.replace(search, '');
      regex = new RegExp('.*' + escapeStringRegexp(originUrl) + '.*$');
      break;
    case 'REGULAR':
      testUrl = href.replace(search, '');
      regex = new RegExp(originUrl);
      break;
    default:
      return false;
  }

  return regex.test(testUrl);
}