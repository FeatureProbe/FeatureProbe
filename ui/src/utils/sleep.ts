const sleep = (delay: number) => {
  return new Promise((res) => {
    setTimeout(() => {
      res('ok');
    }, delay);
  });
};

export default sleep;
