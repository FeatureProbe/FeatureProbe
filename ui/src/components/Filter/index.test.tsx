import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import Filter from '.';
import sleep from 'utils/sleep';
import { IntlWrapper } from 'components/utils/wrapper';

it('Filter snapshot', (done) => {
  (async () => {
    const { asFragment } = render(
      <Filter handleConfirm={jest.fn()} handleClear={jest.fn()} selected>
        <div>
          <div>1</div>
          <div>2</div>
        </div>
      </Filter>
    );
    expect(asFragment()).toMatchSnapshot();
    done();
  })();
});

test('Filter icon click snapshot', (done) => {
  (async () => {
    const { baseElement } = render(
      <Filter handleConfirm={jest.fn()} handleClear={jest.fn()} selected={true}>
        <div>
          <div>1</div>
          <div>2</div>
        </div>
      </Filter>
    , { wrapper: IntlWrapper});

    const ele = document.getElementsByClassName('icon-filter')[0];
    await userEvent.click(ele);

    expect(baseElement).toMatchSnapshot();
    done();
  })();
});

test('Filter Confirm Button click snapshot', (done) => {
  (async () => {
    const mockHandleCancel = jest.fn();
    const mockHandleConfirm = jest.fn();

    const { baseElement } = render(
      <Filter handleConfirm={mockHandleConfirm} handleClear={mockHandleCancel} selected={true}>
        <div>
          <div>1</div>
          <div>2</div>
        </div>
      </Filter>
    , { wrapper: IntlWrapper});

    const ele = document.getElementsByClassName('icon-filter')[0];
    await userEvent.click(ele);
    expect(baseElement).toMatchSnapshot();

    await sleep(1000);

    await userEvent.click(screen.getByText('Clear'));
    expect(mockHandleCancel).toBeCalled();

    await userEvent.click(screen.getByText('Confirm'));
    expect(mockHandleConfirm).toBeCalled();

    expect(baseElement).toMatchSnapshot();

    done();
  })();
});
