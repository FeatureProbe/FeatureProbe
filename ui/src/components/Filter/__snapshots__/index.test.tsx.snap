// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Confirm Button click snapshot 1`] = `
<body>
  <div>
    <span
      class="iconfont icon-filter icon-filter icon-filter-selected"
      customclass="icon-filter icon-filter-selected"
      type="filter"
    />
  </div>
  <div
    style="display: flex; position: absolute; left: 0px; top: 0px; right: 0px; transform: translate(0px, 0px);"
  >
    <div
      class="ui bottom right basic popup transition visible popup"
      style="position: initial;"
    >
      <div>
        <div>
          <div>
            1
          </div>
          <div>
            2
          </div>
        </div>
        <div
          class="popup-footer"
        >
          <span
            class="popup-footer-clear"
          >
            Clear
          </span>
          <span
            class="popup-footer-confirm"
          >
            Confirm
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Filter Confirm Button click snapshot 2`] = `
<body>
  <div>
    <span
      class="iconfont icon-filter icon-filter icon-filter-selected"
      customclass="icon-filter icon-filter-selected"
      type="filter"
    />
  </div>
</body>
`;

exports[`Filter icon click snapshot 1`] = `
<body>
  <div>
    <span
      class="iconfont icon-filter icon-filter icon-filter-selected"
      customclass="icon-filter icon-filter-selected"
      type="filter"
    />
  </div>
  <div
    style="display: flex; position: absolute; left: 0px; top: 0px; right: 0px; transform: translate(0px, 0px);"
  >
    <div
      class="ui bottom right basic popup transition visible popup"
      style="position: initial;"
    >
      <div>
        <div>
          <div>
            1
          </div>
          <div>
            2
          </div>
        </div>
        <div
          class="popup-footer"
        >
          <span
            class="popup-footer-clear"
          >
            Clear
          </span>
          <span
            class="popup-footer-confirm"
          >
            Confirm
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Filter snapshot 1`] = `
<DocumentFragment>
  <span
    class="iconfont icon-filter icon-filter icon-filter-selected"
    customclass="icon-filter icon-filter-selected"
    type="filter"
  />
</DocumentFragment>
`;
